StringIndexOutOfBoundsException in parseApplicationName

```
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
    at java.base/java.lang.String.substring(Unknown Source)
    at com.kruize.environment.kubernetes.KubernetesEnvImpl.parseApplicationName(KubernetesEnvImpl.java:247)
    at com.kruize.environment.kubernetes.KubernetesEnvImpl.getPodMetrics(KubernetesEnvImpl.java:206)
    at com.kruize.environment.kubernetes.KubernetesEnvImpl.insertPodMetrics(KubernetesEnvImpl.java:177)
    at com.kruize.environment.kubernetes.KubernetesEnvImpl.getAllApps(KubernetesEnvImpl.java:164)
    at com.kruize.environment.EnvTypeImpl.getInstance(EnvTypeImpl.java:54)
    at com.kruize.environment.EnvTypeImpl.<clinit>(EnvTypeImpl.java:39)
```


Happens when the pod hash is not within the pod name.

--- a/src/main/java/com/kruize/environment/kubernetes/KubernetesEnvImpl.java
+++ b/src/main/java/com/kruize/environment/kubernetes/KubernetesEnvImpl.java
@@ -244,10 +244,11 @@ public class KubernetesEnvImpl extends EnvTypeImpl
     private static String parseApplicationName(String podName, String hash)
     {
         if (hash != null) {
-            return podName.substring(0, podName.indexOf(hash) - 1);
-        } else {
-            return parseApplicationNameFromInstanceName(podName);
+            int index = podName.indexOf(hash);
+            if (index > 0)
+                return podName.substring(0, index - 1);
         }
+        return parseApplicationNameFromInstanceName(podName);
     }
 
     private void getMonitoringEndpointFromService() throws IOException, ApiException
@@ -285,7 +286,7 @@ public class KubernetesEnvImpl extends EnvTypeImpl
         try {
             URL labelURL = new URL(DeploymentInfo.getMonitoringAgentEndpoint() + "/api/v1/labels");
             String result = HttpUtil.getDataFromURL(labelURL);
-            
+
             if (result.contains("\"pod\"")) {
                 prometheusQuery.setPodLabel("pod");
                 prometheusQuery.setContainerLabel("container");

